lib/execute.js

```diff
@@ -0,0 +1,61 @@
+const { isAsync, log } = require('./util');
+const metaData = Symbol('metaData');
+
+async function executeDescribe(describeBlock) {
+  log(describeBlock.title);
+
+  // Before
+  if (describeBlock.beforeHook) {
+    await executeHook('before', describeBlock.beforeHook);
+  }
+
+  // It blocks
+  for (const ic of describeBlock.itCollection) {
+    await executeItFlow(describeBlock, ic);
+  }
+
+  // After
+  if (describeBlock.afterHook) {
+    await executeHook('after', describeBlock.afterHook);
+  }
+}
+
+async function executeItFlow(describeBlock, singleIt) {
+  // beforeEach
+  for (const be of describeBlock.beforeEachHookCollection) {
+    await executeHook('beforeEach', be);
+  }
+
+  // it
+  await executeIt(singleIt.title, singleIt.fn);
+
+  // AfterEach
+  for (const ae of describeBlock.afterEachHookCollection) {
+    await executeHook('afterEach', ae);
+  }
+}
+
+async function executeIt(title, fn, notLoggingSuccess) {
+  try {
```

Comment:
You could add something like 
```suggestion
  try {
      if(isCallbackFunction(_fn)) {
        fn = util.promisify(_fn);
      }
```

to get a way to easily handle callback style tests (`it('should be handled async', function(done) { .... if(err) {done(err); } else done(); }`).